### PR TITLE
border: Add a way to avoid anti-aliasing border corners.

### PR DESCRIPTION
**Pull request by emilio**
I need this for table border-collapsing
(https://bugzilla.mozilla.org/show_bug.cgi?id=1487407).

I only need solid borders and such, so for now I restricted the implementation
to that since it's way simpler and allowed me to only modify the border_solid
shader.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/3107)

### CODE REVIEW

File: webrender_api/src/display_item.rs [hunk 1]
```diff
@@ -262,6 +262,11 @@ pub struct NormalBorder {
     pub top: BorderSide,
     pub bottom: BorderSide,
     pub radius: BorderRadius,
+    /// Whether to do anti-alias on the border corners.
```
Comment (gw3583) [inline webrender_api/src/display_item.rs]: nit: Whether to apply anti-aliasing ....

**Comment by emilio** [comment]
> What should we do in the case that the border is a transformed (e.g. rotated) element.

I don't think gecko does that, and that'd cause seams on the corners which we don't really want to. So I think respecting the no-aa is fine for now at least.

**Comment by emilio** [comment]
I fixed the nit. I assume this is r=you @gw3583? (Not sure, so given I'm still waiting for some Gecko changes to get review I'd rather wait).

**Comment by Gankro** [comment]
it's a bit weird that you use a bool in the wr API, but in the gecko bindings introduced an enum to represent this?

**Comment by Gankro** [comment]
for reference: https://phabricator.services.mozilla.com/differential/changeset/?ref=155872&whitespace=ignore-most

**Comment by emilio** [comment]
Yeah, I prefer avoiding boolean arguments, but inside WR it's not used as an argument anywhere.

@bors-servo r=kvark

In any case if it becomes confusing later I can change it to make it an enum everywhere.

**Pull request merged**